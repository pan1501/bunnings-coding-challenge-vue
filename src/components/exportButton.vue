<style scoped>
button {
  color: #2c3e50;
  background-color: lightgreen;
  box-sizing: border-box;

  align-items: center;
  justify-content: center;
  vertical-align: middle;
  border: 2px solid;
  cursor: pointer;
  padding: 0 38px;
  height: 56px;
  border-radius: 8px;
  font-size: 16px;
  font-family: Montserrat, Arial, Helvetica, sans-serif;
  font-weight: 600;
  font-style: normal;
  text-decoration: none;
  line-height: 1.5;
  border-color: lightgreen;
}
button:hover {
  background-color: rgb(109, 176, 109);
  border-color: rgb(109, 176, 109);
}
</style>
<template>
  <button v-on:click="exportCSV(output)">Export csv</button>
</template>

<script>
export default {
  name: "exportButton",
  props: {
    output: Array,
  },
  methods: {
    exportCSV(output) {
      let csvTitle = `${Object.keys(output[0])
        .map((value) => value)
        .join(",")}\r\n`;
      let csvContent = "";
      output.forEach((record) => {
        csvContent += `${Object.values(record).join()}\r\n`;
      });

      let csv = csvTitle + csvContent;

      var encodedUri = "data:text/csv;charset=utf-8," + encodeURI(csv);
      var link = document.createElement("a");
      link.href = encodedUri;
      link.download = "output.csv";

      link.click();
    },
  },
};
</script>
