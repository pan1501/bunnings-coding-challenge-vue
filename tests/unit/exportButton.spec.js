import exportButton from "@/components/exportButton.vue";
import { mount } from "@vue/test-utils";
describe("exportButton tests", () => {
  it("button click with correct output", () => {
    const output = [
      {
        SKU: "165-rcy-650",
        Description: "Tea - Decaf 1 Cup",
        Source: "A",
      },
      {
        SKU: "167-eol-949",
        Description: "Cheese - Grana Padano",
        Source: "A",
      },
      {
        SKU: "280-oad-768",
        Description: "Bread - Raisin",
        Source: "A",
      },
      {
        SKU: "647-vyk-317",
        Description: "Walkers Special Old Whiskey",
        Source: "A",
      },
      {
        SKU: "650-epd-782",
        Description: "Carbonated Water - Lemon Lime",
        Source: "A",
      },
      {
        SKU: "999-eol-949",
        Description: "Cheese - Grana Padano",
        Source: "B",
      },
      {
        SKU: "999-epd-782",
        Description: "Carbonated Water - Lemon Lime",
        Source: "B",
      },
    ];
    const wrapper = mount(<exportButton output={output} />);
    expect(wrapper.html()).toContain("Export csv");
    const button = wrapper.find("button");
    console.log(button);
    button.trigger("click");
  });

  it("button click without empty output", () => {
    // expect(wrapper.vm.message).toBe("");
    // const button = wrapper.find("button");
    // button.trigger("click");
    // expect(wrapper.vm.message).toBe("TRY AGAIN");
  });
});
