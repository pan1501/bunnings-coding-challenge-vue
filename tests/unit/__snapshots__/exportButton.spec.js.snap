// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exportButton tests renders exportButton 1`] = `
Object {
  "__v_isVNode": true,
  "__v_skip": true,
  "anchor": null,
  "appContext": null,
  "children": null,
  "component": null,
  "dirs": null,
  "dynamicChildren": null,
  "dynamicProps": null,
  "el": null,
  "key": null,
  "patchFlag": 0,
  "props": Object {
    "output": Array [
      Object {
        "Description": "Tea - Decaf 1 Cup",
        "SKU": "165-rcy-650",
        "Source": "A",
      },
      Object {
        "Description": "Cheese - Grana Padano",
        "SKU": "167-eol-949",
        "Source": "A",
      },
      Object {
        "Description": "Bread - Raisin",
        "SKU": "280-oad-768",
        "Source": "A",
      },
      Object {
        "Description": "Walkers Special Old Whiskey",
        "SKU": "647-vyk-317",
        "Source": "A",
      },
      Object {
        "Description": "Carbonated Water - Lemon Lime",
        "SKU": "650-epd-782",
        "Source": "A",
      },
      Object {
        "Description": "Cheese - Grana Padano",
        "SKU": "999-eol-949",
        "Source": "B",
      },
      Object {
        "Description": "Carbonated Water - Lemon Lime",
        "SKU": "999-epd-782",
        "Source": "B",
      },
    ],
  },
  "ref": null,
  "scopeId": null,
  "shapeFlag": 4,
  "slotScopeIds": null,
  "ssContent": null,
  "ssFallback": null,
  "staticCount": 0,
  "suspense": null,
  "target": null,
  "targetAnchor": null,
  "transition": null,
  "type": Object {
    "methods": Object {
      "exportCSV": [Function],
    },
    "name": "exportButton",
    "props": Object {
      "output": [Function],
    },
    "render": [Function],
  },
}
`;
